app/forms/add_to_order_form.rb

```diff
@@ -163,20 +161,23 @@ def create_cross_core_project_and_add_order!
 
       original_order.update!(cross_core_project: @order_project)
 
-      product_order = Order.find_or_create_by!(
-        facility: product_facility,
-        account_id:,
-        user_id: original_order.user_id,
-        created_by: created_by.id,
-        cross_core_project: @order_project
-      )
+      product_order = find_or_create_product_order
+
+      # If the product that is going to be added requires a merge order, then it is created as part of the merge_order method.
+      # Otherwise, just use product_order.
```

Comment:
```suggestion
      # If the product that is going to be added requires a merge. order, 
      # then we create the merge order as part of the merge_order method.
      # We don't want to purchase the merge_order yet because we don't have 
      # all the detail needed from the user (reservation time, order form, etc)
```